core/src/components/button/button.tsx

```diff
@@ -421,7 +421,7 @@ export class Button implements ComponentInterface, AnchorInterface, ButtonInterf
             <slot></slot>
             <slot name="end"></slot>
           </span>
-          {theme === 'md' && <ion-ripple-effect type={this.rippleType}></ion-ripple-effect>}
+          {(theme === 'md' || theme === 'ionic') && <ion-ripple-effect type={this.rippleType}></ion-ripple-effect>}
```

Comment:
Should the ripple effect only be added when `mode` is `'md'`? I imagine this won't work currently with the one stylesheet.

Reply:
https://github.com/ionic-team/ionic-framework/pull/29754/commits/6a2e0ec3bd67de79970049bbbebcd162987aa25f